### ISSUE TITLE
Add range context support to media-feature-name-* rules

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#1744]
Ref: https://github.com/stylelint/stylelint/pull/1732#issuecomment-235888292

``` css
@media (10px < width > 50px) {}
```

Is dependent on the creation of a [range context](https://www.w3.org/TR/mediaqueries-4/#mq-range-context) parser.

**Comment by vankop** [comment]
I have look only on tests and did not understand from them that this is not a part of the parser yet 🤦🏼‍♂️

**Comment by evilebottnawi** [comment]
@vankop because parser is not maintainable :disappointed:

**Comment by jdiogopeixoto** [comment]
I am available to work on this. 
Should I change media-feature-name-* rules to not ignore range context medias? It would be necessary to make a workaround, since the parser is not being maintained.

**Comment by jeddy3** [comment]
> I am available to work on this.

Fantastic!

> Should I change media-feature-name-* rules to not ignore range context medias?

Yes. If you can adapt the rules to support range context media robustly, then we should no longer ignore them.

**Comment by jdiogopeixoto** [comment]
Nice, I will start work on this.
I noticed that already exists two rules for ranged operators. Because of that wouldn't it be better to create new rules for each `media-feature-name-*` instead of adapting the rules? Just for consistency.

**Comment by jeddy3** [comment]
> I will start work on this.

Good stuff.

> I noticed that already exists two rules for ranged operators. Because of that wouldn't it be better to create new rules for each `media-feature-name-*` instead of adapting the rules?

The two rules (`media-feature-range-operator-space-after` and `media-feature-range-operator-space-before`) are targeting the _range operator_ within _media features_, which is a specific bit of punctuation.

Whereas the `media-feature-name-*` rules are targeting _media feature names_, which are present:

- [outside of the range context](https://drafts.csswg.org/mediaqueries-4/#mq-features)
- [within the range context](https://drafts.csswg.org/mediaqueries-4/#mq-range-context)

At the moment, the `media-feature-name-*` rules only check _media feature names_ outside of the range context. It should, however, check all _media feature names_ regardless of the context.  To do so [would be consistent with our conventions](https://stylelint.io/user-guide/about-rules#about-rule-names).
